src/test/java/teammates/test/cases/action/AdminActivityLogPageActionTest.java

```diff
@@ -1,27 +1,551 @@
+        SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy HH");
+        sdf.setTimeZone(TimeZone.getTimeZone(Const.SystemParams.ADMIN_TIME_ZONE));
+
+        assertTrue(message.contains("Total Logs gone through in last search: " + totalLogs));
+        assertTrue(message.contains("Total Relevant Logs found in last search: " + filteredLogs));
+        assertTrue(message.contains("Logs are from following version(s): 1"));
+        assertTrue(message.contains("All available version(s): 1"));
+        // bug might be introduced when the time is 00:00 AM.
+        assertTrue(message.contains("The earliest log entry checked on <b>" + sdf.format(earliestDateInUtc.getTime())));
+    }
+
+    private void verifyLocalTimeInStatusMessage(String message, Date timeInUtc, double localTimeZone) {
+        SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy HH");
+        Calendar timeLocal = TimeHelper.convertToUserTimeZone(TimeHelper.dateToCalendar(timeInUtc), localTimeZone);
+        assertTrue(message.contains(sdf.format(timeLocal.getTime())));
+        assertTrue(message.contains(String.format("in Local Time Zone (%.1f)", localTimeZone)));
+    }
+
+    private void verifyLoadingLocalTimeAjaxResult(String expected, String role, String googleId, long timeInMillis) {
+        String[] params = new String[]{"logRole", role, "logGoogleId", googleId,
+                                       "logTimeInAdminTimeZone", String.valueOf(timeInMillis)};
+
+        AdminActivityLogPageAction action = getAction(params);
+        AjaxResult result = getAjaxResult(action);
+        AdminActivityLogPageData pageData = (AdminActivityLogPageData) result.data;
+        assertEquals(expected, pageData.getLogLocalTime());
+    }
+
+    private void verifyManyLogs(int totalLogs, int first, int last,
+                                PageData pageData, String statusMessage, Date earliestDateInUtc) {
+        List<ActivityLogEntry> actualLogs = ((AdminActivityLogPageData) pageData).getLogs();
+        int numLogs = last - first + 1;
+
+        verifyLogsIdInRange(actualLogs, first, last);
+        verifyStatusMessage(statusMessage, totalLogs, numLogs, earliestDateInUtc);
+    }
+
+    private void verifyLogsIdInRange(List<ActivityLogEntry> actualLogs, int first, int last) {
+        assertEquals(last - first + 1, actualLogs.size());
+        for (int i = 0; i < actualLogs.size(); i++) {
+            assertEquals(String.format("id4%02d", first + i), actualLogs.get(i).getId());
+        }
+    }
+
+    private void insertLogMessagesAtTime(List<String> msgList, long timeMillis) {
+        insertLogMessageAtTimeWithInterval(msgList, timeMillis, 1);
+    }
+
+    private void insertLogMessageAtTimeWithInterval(List<String> msgList, long timeMillis, int intervalInSecond) {
+        int levelInfo = 1;
+        // bug might be introduced when the time is 00:00 AM.
+        // but this situation is really rare and can be solved by re-running the test case
```

Comment:
If a phrase, start with a lower case letter and don't put a fullstop. If a full sentence, start with an upper case letter and end with a fullstop.